views/default/page/layouts/default.php

```diff
@@ -1,36 +1,58 @@
 <?php
+
 /**
  * Elgg default layout
  *
- * @package Elgg
- * @subpackage Core
+ * @uses $vars['class']        Additional CSS classes to apply to layout
  *
- * @uses $vars['content'] Content string
- * @uses $vars['class']   Additional class to apply to layout
- * @uses $vars['nav']     Optional override of the page nav (default: breadcrumbs)
- * @uses $vars['title']   Optional title for main content area
- * @uses $vars['header']  Optional override for the header
- * @uses $vars['footer']  Optional footer
- */
+ * @uses $vars['breadcrumbs']  An array of breadcrumbs (see navigation/breadcrumbs)
+ * @uses $vars['nav']          Optional page navigation (replaces navigation/breadcrumbs)
+ *
+ * @uses $vars['title']        Optional title for main content area
+ * @uses $vars['header']       Optional override for the header
+ *
+ * @uses $vars['content']      Content string
 
-$class = 'elgg-layout elgg-layout-default clearfix';
-if (isset($vars['class'])) {
-	$class = "$class {$vars['class']}";
+ * @uses $vars['footer']       Optional footer
+ *
+ * @uses $vars['sidebar']      Sidebar HTML (default: empty string)
+ *                             Will not be rendered if the value is 'false'
+ * @uses $vars['sidebar_alt']  Second sidebar HTML (default: false)
+ *                             Will not be rendered if the value is 'false'
+ *
+ * @uses $vars['filter']       An optional array of filter tabs
+ *                             Array items should be suitable for usage with
+ *                             elgg_register_menu_item()
+ * @uses $vars['filter_id']    An optional ID of the filter
+ *                             If provided, plugins can adjust filter tabs menu
+ *                             via 'register, menu:filter:$filter_id' hook
+ * @uses $vars['filter_value'] Optional name of the selected filter tab
+ *                             If not provided, will be determined by current page's URL
+ */
+$class = elgg_extract_class($vars, [
+	'elgg-layout',
+	'clearfix'
+]);
+
+$vars['sidebar'] = elgg_extract('sidebar', $vars, '');
+$vars['sidebar_alt'] = elgg_extract('sidebar_alt', $vars, false);
+
+$sidebar = elgg_view('page/layouts/elements/sidebar', $vars);
+$sidebar_alt = '';
+if ($sidebar) {
+	$sidebar_alt = elgg_view('page/layouts/elements/sidebar_alt', $vars);
 }
 
-echo "<div class=\"$class\">";
-
-echo elgg_extract('nav', $vars, elgg_view('navigation/breadcrumbs'));
-
-echo elgg_view('page/layouts/elements/header', $vars);
-
-echo $vars['content'];
-
-// @deprecated 1.8
-if (isset($vars['area1'])) {
-	echo $vars['area1'];
+if ($sidebar && $sidebar_alt) {
+	$class[] = 'elgg-layout-two-sidebar';
+} else if ($sidebar) {
+	$class[] = 'elgg-layout-one-sidebar';
+} else {
+	$class[] = 'elgg-layout-one-column';
 }
 
-echo elgg_view('page/layouts/elements/footer', $vars);
+$body = elgg_view('page/layouts/elements/body', $vars);
 
-echo '</div>';
+echo elgg_format_element('div', [
+	'class' => $class,
+], $sidebar . $sidebar_alt . $body);
```

Comment:
In Aalborg, it's `$sidebar_alt . $body . $sidebar`, but if we are switching to flexbox, we can just use `order`